v2/commands/job/logs.go

```diff
@@ -67,45 +72,66 @@ func Logs(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("Error: unable to write header")
 	}
 
-	for {
-		// write rows as the batches come in up to the limit
-		if limit > 0 && limit <= total {
-			break
-		}
-
-		res, err := logs(ctx, runner.Client, p.ProjectId, p.TaskId, jobId, pageIndex)
-		if err != nil {
-			return err
-		}
-
-		currentPage := len(res.Items)
-		if total+currentPage >= limit {
-			currentPage = limit - total
-		}
+	latestLogTime := time.Time{}
+	sleepRetries := 1
 
-		// only write up to the current page limit
-		if limit > 0 {
-			err = writeRows(res.Items[:currentPage], os.Stdout)
-		} else {
-			err = writeRows(res.Items, os.Stdout)
-		}
-		if err != nil {
-			return err
+	for {
+		for {
+			// write rows as the batches come in up to the limit
+			if limit > 0 && limit <= total {
+				break
+			}
+
+			res, err := logs(ctx, runner.Client, p.ProjectId, p.TaskId, jobId, pageIndex)
+			if err != nil {
+				return err
+			}
+
+			currentPage := len(res.Items)
+			if limit > 0 && total+currentPage >= limit {
+				currentPage = limit - total
+			}
+
+
+			// only write up to the current page limit
+			if limit > 0 {
+				err = writeRows(res.Items[:currentPage], os.Stdout, time.Time{})
+			} else {
+				err = writeRows(res.Items, os.Stdout, latestLogTime)
+			}
+			if err != nil {
+				return err
+			}
+
+			printedLogTime := time.Time(res.Items[currentPage-1].Timestamp.Value)
+			if printedLogTime.After(latestLogTime) {
+				latestLogTime = printedLogTime
+			} else {
+				fmt.Printf("no new logs from query, sleeping 5s... (x%d)\n", sleepRetries)
```

Comment:
Happy to remove/replace this if needed, but imo nice to have some visual feedback when no logs are updating

Reply:
```suggestion
				fmt.Fprintf(os.Stderr, "no new logs from query, sleeping... (x%d)\n", sleepRetries)
```

Let's:

1. Write this message to stderr so can be treated differently than anything piped to grep on stdout.
1. Remove `5s` since it's not guaranteed to be `common.LogsQueryInterval` and is short enough the user can intuit how long the sleep is